### ISSUE TITLE
[Kubevirt] virt-launcher pod stuck in Terminating after node power cycle

### ISSUE THREAD

**Issue by gdiwate** [kubernetes/kubernetes#122960]
### What happened?

- Created a centos VM using the kubevirt with a disk as block device.
- Rebooted the Node on which the virt-launcher pod is running.
- Pod goes in Terminating state.
- All nodes were in healthy state.
- User kubernetes version v1.27.7.
- Found below error in the kubelet log. Kubelet tries to unmount the path which is not exists.
```
Jan 16 01:43:40 appserv56 kubelet[6715]: E0116 01:43:40.722571    6715 nestedpendingoperations.go:348] Operation for "{volumeName:kubernetes.io/csi/dcx.csi.diamanti.com^pvc-6a70e1b4-8b7f-435a-bf5c-fba2fddea7fe podName:2fb65628-7467-45d4-8dc1-4f77e5cc64db nodeName:}" failed. No retries permitted until 2024-01-16 01:45:42.722524679 -0800 PST m=+15150.170124605 (durationBeforeRetry 2m2s). Error: UnmapVolume.UnmapBlockVolume failed for volume "pvc-6a70e1b4-8b7f-435a-bf5c-fba2fddea7fe" (UniqueName: "kubernetes.io/csi/dcx.csi.diamanti.com^pvc-6a70e1b4-8b7f-435a-bf5c-fba2fddea7fe") pod "2fb65628-7467-45d4-8dc1-4f77e5cc64db" (UID: "2fb65628-7467-45d4-8dc1-4f77e5cc64db") : blkUtil.DetachFileDevice failed. globalUnmapPath:, podUID: 2fb65628-7467-45d4-8dc1-4f77e5cc64db, bindMount: true: failed to unmap device from map path. mapPath is empty
Jan 16 01:45:42 appserv56 kubelet[6715]: I0116 01:45:42.800589    6715 reconciler_common.go:172] "operationExecutor.UnmountVolume started for volume \"pvc-6a70e1b4-8b7f-435a-bf5c-fba2fddea7fe\" (UniqueName: \"kubernetes.io/csi/dcx.csi.diamanti.com^pvc-6a70e1b4-8b7f-435a-bf5c-fba2fddea7fe\") pod \"2fb65628-7467-45d4-8dc1-4f77e5cc64db\" (UID: \"2fb65628-7467-45d4-8dc1-4f77e5cc64db\") "
Jan 16 01:45:42 appserv56 kubelet[6715]: W0116 01:45:42.800664    6715 volume_path_handler_linux.go:62] couldn't find loopback device which takes file descriptor lock. Skip detaching device. device path: "2fb65628-7467-45d4-8dc1-4f77e5cc64db"
Jan 16 01:45:42 appserv56 kubelet[6715]: W0116 01:45:42.800689    6715 volume_path_handler.go:217] Warning: Unmap skipped because symlink does not exist on the path: /var/lib/kubelet/pods/2fb65628-7467-45d4-8dc1-4f77e5cc64db/volumeDevices/kubernetes.io~csi/pvc-6a70e1b4-8b7f-435a-bf5c-fba2fddea7fe
```
```
$ kgp                                                                                                                                            
NAME                         READY   STATUS        RESTARTS   AGE   IP       NODE        NOMINATED NODE   READINESS GATES                                                  
virt-launcher-centos-rnkvz   0/1     Terminating   0          51m      appserv92              1/1                           
```
- **Workaround**: Restart the kubelet service or delete the pod forcefully.

### What did you expect to happen?

The virt-launcher po should be in running state.

### How can we reproduce it (as minimally and precisely as possible)?

1. Create centos VM with 1 disk(block device) using below spec on kubevirt. 
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: centos-disk
spec:
  accessModes:
    - ReadWriteOnce
  volumeMode: Block
  resources:
    requests:
      storage: 100Gi
---
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine                                                                                                                                                       
metadata:                                                                                                                                                                  
  name: centos                                                                                                                                                             
spec:                                                                                                                                                                      
  running: false                                                                                                                                                           
  template:                                                                                                                                                                
    metadata:                                                                                                                                                              
      annotations:                                                                                                                                                         
        diamanti.com/endpoint0: '{"network":"default","perfTier":"high"}'                                                                                                  
      labels:                                                                                                                                                              
        kubevirt.io/domain: centos                                                                                                                                         
    spec:                                                                                                                                                                  
      domain:                                                                                                                                                              
        ioThreadsPolicy: auto                                                                                                                                              
        cpu:                                                                                                                                                               
          cores: 8                                                                                                                                                         
        devices:                                                                                                                                                           
          blockMultiQueue: true                                                                                                                                            
          disks:                                                                                                                                                           
          - disk:                                                                                                                                                          
              bus: virtio                                                                                                                                                  
            name: harddrive                                                                                                                                                
          - disk:                                                                                                                                                          
              bus: virtio                                                                                                                                                  
            name: harddrive2                                                                                                                                               
            dedicatedIOThread: true                                                                                                                                        
        machine:                                                                                                                                                           
          type: q35                                                                                                                                                        
        resources:                                                                                                                                                         
          requests:                                                                                                                                                        
            memory: 16G                                                                                                                                                    
      volumes:                                                                                                                                                             
      - name: harddrive                                                                                                                                                    
        persistentVolumeClaim:                                                                                                                                             
          claimName: centos-os                                                                                                                                             
      - name: harddrive2                                                                                                                                                   
        persistentVolumeClaim:                                                                                                                                             
          claimName: centos-disk
```
2. Restart the node on which the virt-launcher pod is running.
3. Wait till the node is up and goos state.
4. Observe the pod status, it would be in terminating state.
5. Try step 1-4 if not reproduced.
6. See the kubelet logs.

### Anything else we need to know?

There was the same issue resolved for the nfs mount https://github.com/kubernetes/kubernetes/issues/114546

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.7", GitCommit:"07a61d861519c45ef5c89bc22dda289328f29343", GitTreeState:
"clean", BuildDate:"2023-10-18T11:42:32Z", GoVersion:"go1.20.10", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.7", GitCommit:"07a61d861519c45ef5c89bc22dda289328f29343", GitTreeState:
"clean", BuildDate:"2023-10-18T11:33:23Z", GoVersion:"go1.20.10", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME="Rocky Linux"
VERSION="8.6 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.6 (Green Obsidian)"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
ROCKY_SUPPORT_PRODUCT="Rocky Linux"
ROCKY_SUPPORT_PRODUCT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8"

$ uname -a
Linux appserv64 4.18.0-372.9.1.el8.x86_64 #1 SMP Tue May 10 14:48:47 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>
```
$ crio version
INFO[2024-01-25 02:02:40.557038574-08:00] Starting CRI-O, version: 1.25.4, git: unknown(clean) 
Version:        1.25.4
GitCommit:      unknown
GitCommitDate:  unknown
GitTreeState:   clean
BuildDate:      2023-08-10T08:50:26Z
GoVersion:      go1.19
Compiler:       gc
Platform:       linux/amd64
Linkmode:       dynamic
BuildTags:      
  rpm_crashtraceback
  exclude_graphdriver_btrfs
  btrfs_noversion
  exclude_graphdriver_devicemapper
  libdm_no_deferred_remove
  seccomp
  containers_image_openpgp
LDFlags:          -X github.com/cri-o/cri-o/internal/pkg/criocli.DefaultsPath= -X  github.com/cri-o/cri-o/internal/version.buildDate=2023-08-10T08:50:26Z -X  github.com/cri-o/cri-o/internal/version.gitCommit=3ad059f673902a0b502da5e9ec80714df08f3f4e -X  github.com/cri-o/cri-o/internal/version.version=1.25.4 -X  github.com/cri-o/cri-o/internal/version.gitTreeState=clean  -B 0x0ae0f03614d4cdbdf46118d31839208bbeb1c635 -extldflags '-Wl,-z,relro  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld ' -compressdwarf=false
SeccompEnabled:   true
AppArmorEnabled:  false
Dependencies:     
```
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by kannon92** [comment]
From the logs, looks to be storage related.
/sig node storage

Can you see if any events are giving more hints?

**Comment by aryasoni98** [comment]
@gdiwate @kannon92 @k8s-ci-robot Can i work on this issue .

**Comment by kannon92** [comment]
@aryasoni98 please look at other issues. We aren't sure if this is a bug yet.

**Comment by gnufied** [comment]
Have you tried to run with a fixed version of 1.27 which uses https://github.com/kubernetes/kubernetes/pull/122397 ?
